src/main/java/org/casbin/spring/boot/autoconfigure/CasbinAutoConfiguration.java

```diff
@@ -83,27 +86,23 @@ public Adapter autoConfigJdbcAdapter(
             @CasbinDataSource ObjectProvider<DataSource> casbinDataSource,
             JdbcTemplate jdbcTemplate,
             CasbinProperties properties,
-            CasbinExceptionProperties exceptionProperties
+            CasbinExceptionProperties exceptionProperties,
+            DataSourceProperties dataSourceProperties
     ) {
         JdbcTemplate jdbcTemplateToUse = getJdbcTemplate(jdbcTemplate, casbinDataSource);
         String databaseName = getDatabaseName(jdbcTemplateToUse.getDataSource());
         CasbinDataSourceInitializationMode initializeSchema = properties.getInitializeSchema();
         boolean autoCreateTable = initializeSchema == CasbinDataSourceInitializationMode.CREATE;
         String tableName = properties.getTableName();
         logger.info("Casbin current use database product: {}", databaseName);
-        switch (databaseName) {
-            case "mysql":
-            case "h2":
-                return new JdbcAdapter(jdbcTemplateToUse, exceptionProperties, tableName, autoCreateTable);
-            case "postgresql":
-                return new PostgreSQLAdapter(jdbcTemplateToUse, exceptionProperties, tableName, autoCreateTable);
-            case "oracle":
-                return new OracleAdapter(jdbcTemplateToUse, exceptionProperties, tableName, autoCreateTable);
-            case "db2":
-                return new DB2Adapter(jdbcTemplateToUse, exceptionProperties, tableName, autoCreateTable);
-            default:
-                throw new CasbinAdapterException("Can't find " + databaseName + " jdbc adapter");
+        try {
+            return new JDBCAdapter(dataSourceProperties.getDriverClassName(), dataSourceProperties.getUrl(),
+                    dataSourceProperties.getUsername(), dataSourceProperties.getPassword(),
+                    exceptionProperties.isRemovePolicyFailed(), tableName, autoCreateTable);
+        } catch (Exception ignored) {
         }
+
+        throw new CasbinAdapterException("Can't create JDBCAdapter, because some properties are not right");
```

Comment:
This doesn't help in tracking errors. Either output specific properties or throw the native Exception